pacman/README.md

```diff
@@ -0,0 +1,155 @@
+
+### Deliverable 2/2 - Report
+
+You need to provide a report in PDF format of 4 pages max (not including appendices for results). Your report must (i) contains the formalization of the Part 1 as a search problem, (ii) describes and justify carefully the (admissible) heuristics you have considered, (iii) show and discuss benchmark results, comparing the performance of your agent with the other algorithms you have implemented.
+
+### Deadline
+
+Due to Oct 28,2018. Submit your deliverables [there](https://submit.montefiore.ulg.ac.be/teacher/courseDetails/INFO8006/)
+
+## Ghosts (Part 2/3)
+
+Coming soon
+
+## Blinkwalls (Part 3/3)
+
+Coming soon
+
+## Computation time budget (All parts)
+
+The computation time of the agent is limited on-game. More specifically, the method ```get_action``` of the agent prematurely terminates when the computation time transgress the given `timeout` parameter (fixed to 60 seconds during evaluation process). On the other side, when the option ``` --enable-search-before-game ``` is specified, the method ``` registerInitialState ``` is called before the game starts, enabling computation without any computation time limit. Both configurations are considered during evaluation process.
+
+## Illegal moves (All parts)
+
+You need to ensure that your agent always returns a legal move. If it is not the case, the game engine ignores it and repeat the previous move as long as it remains legal. Otherwise, it just keep getting stuck in the same location.
+
+
+
+## Evaluation
+
+The evaluation of your deliverables is based on the following criteria :
+
+- Clarity of your report
+	* Avoid long and vague sentences and be straight to the point. Ideally, the length of each sentence should not exceed a line.
+	* Do not hesitate to illustrate with examples.
+	* Do not overload your graphics, and write short and clear captions. The reader should understand them in a quick look.
+	* Lack of structure leads to lack of clarity. We recommend you to design your report outline on top of the one described in the previous section.
+- Clarity and structure of the source code
+	* Avoid single-file long code source, and prefers to use a multiple-file modular architecture. 
+	* Name your variables-attributes-classes according to their usage. Comment your code so that explanations are concise and clear enough to allow the reader to understand the semantics in a quick look.
+
+These criterions are all importants. We particularly value well written reports and code documentation. 
+
+:warning: Plagiarism is checked and sanctioned by a 0 grade. It is fine to reuse external code, as long as (i) it does not cover your whole approach, (ii) you fully understand the principles behind and (iii) you do give credits in your code.
```

Comment:
We do not allow to reuse external code. The projects should be simple enough for this to not be necessary.